src/azul/tdr.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import re
+from typing import List
+from typing import NamedTuple
+
+from boltons.cacheutils import cachedproperty
+from more_itertools import one
+import requests
+
+from azul.bigquery import GoogleQueryAdapter
+from azul.indexer import BundleFQID
+
+log = logging.getLogger(__name__)
+
+
+class Snapshot(NamedTuple):
+    uuid: str
+    name: str
+
+
+class AzulTDRClient:
+    valid_uuid_re = re.compile(r'[-0-9a-z]*')
+
+    @cachedproperty
+    def query_service(self) -> GoogleQueryAdapter:
+        return GoogleQueryAdapter.create_adaptee()
+
+    def __init__(self, terra_endoint: str):
+        self.endpoint = terra_endoint
+
+    def get_snapshot(self) -> Snapshot:
+        response = requests.get(self.endpoint +
```

Comment:
I want us to move away from requests and use urllib3 directly.

And see https://urllib3.readthedocs.io/en/latest/user-guide.html#query-parameters for a easier to parameterize way to pass query args.